integrated_channels/moodle/client.py

```diff
@@ -311,33 +312,175 @@ def _wrapped_create_course_completion(self, user_id, payload):
+        """
+        params = {
+            'wsfunction': 'mod_forum_get_forum_discussions',
+            'forumid': forum_id,
+            'moodlewsrestformat': 'json',
+            'wstoken': self.token,
+        }
+        return self._post(params)
+
+    def get_announcement_post(self, forum_id):
+        """
+        Returns the id of the discussion post matching our announcement or returns nothing.
+        """
+        response = self._get_forum_discussions(forum_id)
+        response_text = json.loads(response.text)
+        for _, discussion in enumerate(response_text['discussions']):
+            if discussion['subject'] == ANNOUNCEMENT_POST_SUBJECT:
+                return discussion['id']
+        return ''
+
+    def _update_announcement_post(self, course_id, announcement):
+        """
+        Updates announcement post with announcement text composed in the exporter.
+        """
+        forum_id = self.get_announcement_forum(course_id)
+        post_id = self.get_announcement_post(forum_id)
+        params = {
+            'wsfunction': 'mod_forum_update_discussion_post',
+            'postid': post_id,
+            'message': announcement,
+        }
+        return self._post(params)
+
+    def _create_forum_post(self, course_id, announcement):
+        """
+        Creates a new Announcement post featuring edX course details and link.
+        """
+        forum_id = self.get_announcement_forum(course_id)
+        params = {
+            'wsfunction': 'mod_forum_add_discussion',
+            'forumid': forum_id,
+            'subject': ANNOUNCEMENT_POST_SUBJECT,
+            'options[0][name]': 'discussionpinned',
+            'options[0][value]': 'true',
+            'message': announcement
+        }
+        return self._post(params)
+
+    def create_content_metadata(self, serialized_data):
+        """
+        This content metadata creation has 3 phases:
+        1. Create the course in Moodle with most information
+        2. Create announcement post with course summary and details
+        3. Update the course format to "singleactivity" to show course details.
+        """
+        serialized_data['wsfunction'] = 'core_course_create_courses'
+        announcement = serialized_data.pop('courses[0][announcement]')
+        course_format = serialized_data.pop('courses[0][format]')
+        response = self._wrapped_create_content_metadata(serialized_data)
+        # Response format should be [{"id":1, "shortname": "name"},{...}]
+        if response[0].get('id', None):
+            post = self._create_forum_post(
+                response[0].get('id'),
+                announcement
+            )
+            if post.json().get('warnings', None) or post.json().get('exception', None):
+                params = {
+                    'wsfunction': 'core_course_delete_courses',
+                    'courseids[]': response[0].get('id')
+                }
+                self._post(params)
```

Comment:
Yes this is just getting called and hoping for the best.
This is because I don't know what to do if delete specifically fails. We need to delete the course to have any hope of getting back to a clean state, otherwise:
- If the course is in Moodle, but we don't register it, we're just going to hit errors constantly.
- If we do register it, but never made the announcement post successfully, then it's in an incomplete state on our side and nothing will ever trigger an update (even if the course itself changes, the updates for the announcement post check the existing post id, which won't exist).

So if delete fails, I really don't know what to do here. We can't do any clean up and the errors will just keep coming.